src/handlers/flow/FlowStreamHandler.h

```diff
@@ -0,0 +1,222 @@
+    uint32_t if_out_index;
+};
+
+struct FlowPacket {
+    timespec stamp;
+    std::vector<FlowData> flow_data;
+
+    FlowPacket(timespec stamp)
+        : stamp(stamp)
+    {
+    }
+};
+
+class FlowMetricsBucket final : public visor::AbstractMetricsBucket
+{
+
+protected:
+    mutable std::shared_mutex _mutex;
+
+    Cardinality _srcIPCard;
+    Cardinality _dstIPCard;
+
+    TopN<std::string> _topGeoLoc;
+    TopN<std::string> _topASN;
+
+    struct topns {
+        TopN<std::string> topSrcIP;
+        TopN<std::string> topDstIP;
+        TopN<uint16_t> topSrcPort;
+        TopN<uint16_t> topDstPort;
+        TopN<uint32_t> topInIfIndex;
+        TopN<uint32_t> topOutIfIndex;
+        topns(std::string metric)
+            : topSrcIP("flows", "ip", {"top_src_ip_" + metric}, "Top source IP addresses by " + metric)
+            , topDstIP("flows", "ip", {"top_dst_ip_" + metric}, "Top destination IP addresses by " + metric)
+            , topSrcPort("flows", "port", {"top_src_ports_" + metric}, "Top source ports by " + metric)
+            , topDstPort("flows", "port", {"top_dst_ports_" + metric}, "Top destination ports by " + metric)
+            , topInIfIndex("flows", "index", {"top_in_if_index_" + metric}, "Top input interface indexes  by " + metric)
+            , topOutIfIndex("flows", "index", {"top_out_if_index_" + metric}, "Top output interface indexes by " + metric)
+        {
+        }
+    };
+
+    topns _topByBytes;
+    topns _topByPackets;
+
+    // total numPackets is tracked in base class num_events
+    struct counters {
+        Counter UDP;
+        Counter TCP;
+        Counter OtherL4;
+        Counter IPv4;
+        Counter IPv6;
+        Counter total;
+        counters()
+            : UDP("flows", {"udp"}, "Count of UDP packets")
+            , TCP("flows", {"tcp"}, "Count of TCP packets")
+            , OtherL4("flows", {"other_l4"}, "Count of packets which are not UDP or TCP")
+            , IPv4("flows", {"ipv4"}, "Count of IPv4 packets")
+            , IPv6("flows", {"ipv6"}, "Count of IPv6 packets")
+            , total("flows", {"flows"}, "Count of total flows")
+        {
+        }
+    };
+    counters _counters;
+
+    Quantile<std::size_t> _payload_size;
+
+    Rate _rate;
+    Rate _throughput;
+
+public:
+    FlowMetricsBucket()
+        : _srcIPCard("flows", {"cardinality", "src_ips_in"}, "Source IP cardinality")
+        , _dstIPCard("flows", {"cardinality", "dst_ips_out"}, "Destination IP cardinality")
+        , _topGeoLoc("flows", "geo_loc", {"top_geoLoc"}, "Top GeoIP locations")
+        , _topASN("flows", "asn", {"top_ASN"}, "Top ASNs by IP")
+        , _topByBytes("bytes")
+        , _topByPackets("packets")
+        , _payload_size("flows", {"payload_size"}, "Quantiles of payload sizes, in bytes")
+        , _rate("flows", {"rates", "pps"}, "Rate of flow packets per second")
+        , _throughput("payload", {"rates", "bps"}, "Rate of flow packets size in bytes per second")
```

Comment:
hm this is size of the flow packet itself, not the inner flow data right? Maybe we can make that more clear.

Reply:
so this is the rate of all _combined_ flow inner data? then it should say "Rate of all combined flows in bytes per second"